pkg/auth/handler/webapp/viewmodels/base.go

```diff
@@ -239,9 +240,12 @@ func (m *BaseViewModeler) ViewModel(r *http.Request, rw http.ResponseWriter) Bas
 		bpLang = intl.ResolveRecaptchaV2(resolvedLanguageTag)
 	}
 
+	requestURI := r.URL.RequestURI()
+	hasXStep := strings.Contains(requestURI, webapp.AuthflowQueryKey)
```

Comment:
This may result in false positive (but that's is unlikely though). 

I would write

```
HasXStep: r.URL.Query().Has(webapp.AuthflowQueryKey),
```